renegade-crypto/src/hash.rs

```diff
@@ -12,6 +12,31 @@ use crate::constants::{POSEIDON_MDS_MATRIX_T_3, POSEIDON_ROUND_CONSTANTS_T_3};
 /// A type alias for the arkworks Poseidon sponge over the Stark field
 pub type PoseidonParams = PoseidonConfig<Scalar::Field>;
 
+/// A hash chain from a seed used to compute CSPRNG values
+pub struct PoseidonCSPRNG {
+    /// The seed of the CSPRNG, this is chained into a hash function
+    /// to give pseudorandom values
+    seed: Scalar,
```

Comment:
nit: can we call this `state`, and update the doc comment to mention that it is initially set to the seed? makes the iterator functionality read a bit more like its expected to